pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -949,7 +949,7 @@ func updateSnapshotsInAIStatus(status *types.AppInstanceStatus, config types.App
 		}
 		status.SnapStatus.RequestedSnapshots = append(status.SnapStatus.RequestedSnapshots, newSnapshotStatus)
 	}
-	status.SnapStatus.SnapshotsToBeDeleted = snapshotsToBeDeleted
+	status.SnapStatus.SnapshotsToBeDeleted = append(status.SnapStatus.SnapshotsToBeDeleted, snapshotsToBeDeleted...)
```

Comment:
Where/when do you clear SnapStatus.SnapshotsToBeDeleted?

Reply:
Oooor.... Actually, the list created here https://github.com/lf-edge/eve/blob/5bd2b88a46ac7a72a430db68f11dccc67c84af93/pkg/pillar/cmd/zedmanager/zedmanager.go#L876 should not lose snapshots before they are really removed... I'll try to reproduce the situation when I saw some issue with the removal and check, how it happens. 
Thanks for pointing to the suspicious place.

Reply:
I fixed the list cleanup. I have added the removal to the triggerSnapshotDeleteion, as mentioned before.